pkg/operator/operator.go

```diff
@@ -90,47 +89,7 @@ type Operator struct {
 }
 
 func NewOperator(ctx context.Context, operator *operator.Operator) (context.Context, *Operator) {
-	config := &aws.Config{
-		STSRegionalEndpoint: endpoints.RegionalSTSEndpoint,
-	}
-
-	// prometheusv1.WithPrometheusMetrics is used until the upstream aws-sdk-go or aws-sdk-go-v2 supports
-	// Prometheus metrics for client-side metrics out-of-the-box
-	// See: https://github.com/aws/aws-sdk-go-v2/issues/1744
-	sess := prometheusv1.WithPrometheusMetrics(WithUserAgent(session.Must(session.NewSession(
-		request.WithRetryer(
-			config,
-			awsclient.DefaultRetryer{NumMaxRetries: awsclient.DefaultRetryerMaxNumRetries},
-		),
-	))), crmetrics.Registry)
-
-	if *sess.Config.Region == "" {
-		log.FromContext(ctx).V(1).Info("retrieving region from IMDS")
-		region, err := ec2metadata.New(sess).Region()
-		*sess.Config.Region = lo.Must(region, err, "failed to get region from metadata server")
-	}
-	ec2api := ec2.New(sess)
-	if err := CheckEC2Connectivity(ctx, ec2api); err != nil {
-		log.FromContext(ctx).Error(err, "ec2 api connectivity check failed")
-		os.Exit(1)
-	}
-	log.FromContext(ctx).WithValues("region", *sess.Config.Region).V(1).Info("discovered region")
-	clusterEndpoint, err := ResolveClusterEndpoint(ctx, eks.New(sess))
-	if err != nil {
-		log.FromContext(ctx).Error(err, "failed detecting cluster endpoint")
-		os.Exit(1)
-	} else {
-		log.FromContext(ctx).WithValues("cluster-endpoint", clusterEndpoint).V(1).Info("discovered cluster endpoint")
-	}
-	// We perform best-effort on resolving the kube-dns IP
-	kubeDNSIP, err := KubeDNSIP(ctx, operator.KubernetesInterface)
-	if err != nil {
-		// If we fail to get the kube-dns IP, we don't want to crash because this causes issues with custom DNS setups
-		// https://github.com/aws/karpenter-provider-aws/issues/2787
-		log.FromContext(ctx).V(1).Info(fmt.Sprintf("unable to detect the IP of the kube-dns service, %s", err))
-	} else {
-		log.FromContext(ctx).WithValues("kube-dns-ip", kubeDNSIP).V(1).Info("discovered kube dns")
-	}
+	sess, ec2api, clusterEndpoint, kubeDNSIP := resolveOperatorProperties(ctx, operator)
```

Comment:
QQ: This wraps stuff that gets generated inside of this constructor but realistically I think this only adds a function signature to the implementation here. Does this help the internal implementation here in other ways?

Reply:
Nope, just make the function shorter for better reading.